api/tests/unit/app_analytics/test_unit_app_analytics_views.py

```diff
@@ -99,6 +105,189 @@ def test_get_usage_data(mocker, admin_client, organisation):
     mocked_get_usage_data.assert_called_once_with(organisation)
 
 
+@pytest.mark.freeze_time("2024-04-30T09:09:47.325132+00:00")
+def test_get_usage_data__current_billing_period(
+    mocker: MockerFixture,
+    admin_client_new: APIClient,
+    organisation: Organisation,
+) -> None:
+    # Given
+    url = reverse("api-v1:organisations:usage-data", args=[organisation.id])
+    url += "?period=current_billing_period"
+
+    mocked_get_usage_data = mocker.patch(
+        "app_analytics.analytics_db_service.get_usage_data_from_influxdb",
+        autospec=True,
+        return_value=[
+            UsageData(flags=10, day=date.today()),
+            UsageData(flags=10, day=date.today() - timedelta(days=1)),
+        ],
+    )
+
+    now = timezone.now()
+    week_from_now = now + timedelta(days=7)
+    four_weeks_ago = now - timedelta(days=28)
+
+    OrganisationSubscriptionInformationCache.objects.create(
+        organisation=organisation,
+        current_billing_term_starts_at=four_weeks_ago,
+        current_billing_term_ends_at=week_from_now,
+        allowed_30d_api_calls=1_000_000,
+    )
+
+    # When
+    response = admin_client_new.get(url)
+
+    # Then
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json() == [
+        {
+            "flags": 10,
+            "day": str(date.today()),
+            "identities": 0,
+            "traits": 0,
+            "environment_document": 0,
+        },
+        {
+            "flags": 10,
+            "day": str(date.today() - timedelta(days=1)),
+            "identities": 0,
+            "traits": 0,
+            "environment_document": 0,
+        },
+    ]
+
+    mocked_get_usage_data.assert_called_once_with(
+        organisation_id=organisation.id,
+        environment_id=None,
+        project_id=None,
+        date_start="-28d",
+        date_stop="now()",
+    )
+
+
+@pytest.mark.freeze_time("2024-04-30T09:09:47.325132+00:00")
+def test_get_usage_data__previous_billing_period(
+    mocker: MockerFixture,
+    admin_client_new: APIClient,
+    organisation: Organisation,
+) -> None:
+    # Given
+    url = reverse("api-v1:organisations:usage-data", args=[organisation.id])
+    url += "?period=previous_billing_period"
+
+    mocked_get_usage_data = mocker.patch(
+        "app_analytics.analytics_db_service.get_usage_data_from_influxdb",
+        autospec=True,
+        return_value=[
+            UsageData(flags=10, day=date.today()),
+            UsageData(flags=10, day=date.today() - timedelta(days=1)),
```

Comment:
I get that this is just getting passed through and asserted in the json below, but we should probably keep this realistic and use dates that would fall in the period we're filtering on.

Reply:
Yeah good call. I normally do things like that but I'm not sure why I didn't line up the dates this time. Updated.